blueprints/python/nbest-transcriptions-example-bot/lexv2-nbest-transcriptions.py

```diff
@@ -0,0 +1,194 @@
+    is not accurate.
+    """
+
+    transcriptions = intent_request['transcriptions']
+
+    if intent_request['invocationSource'] == 'DialogCodeHook':
+        # Disambiguate if there are multiple transcriptions and the top transcription
+        # confidence is below a threshold (0.8 here)
+        if len(transcriptions) > 1 and transcriptions[0]['transcriptionConfidence'] < 0.8:
+            if transcriptions[0]['resolvedSlots'] is not {} and 'Name' in transcriptions[0]['resolvedSlots'] and \
+                    transcriptions[0]['resolvedSlots']['Name'] is not None:
+                return prompt_for_name(intent_request)
+            elif transcriptions[0]['resolvedSlots'] is not {} and 'BirthMonth' in transcriptions[0]['resolvedSlots'] and \
+                    transcriptions[0]['resolvedSlots']['BirthMonth'] is not None:
+                return validate_month(intent_request)
+
+    return continue_conversation(intent_request)
+
+
+def prompt_for_name(intent_request):
+    """
+    If the confidence for name is not high enough, re prompt the user with the recognized names
+    so it can be confirmed.
+    """
+    resolved_names = []
+    for transcription in intent_request['transcriptions']:
+        if transcription['resolvedSlots'] is not {} and 'Name' in transcription['resolvedSlots'] and \
+                transcription['resolvedSlots']['Name'] is not None:
+            resolved_names.append(transcription['resolvedSlots']['Name']['value']['originalValue'])
+    if len(resolved_names) > 1:
+        session_attributes = get_session_attributes(intent_request)
+        slots = get_slots(intent_request)
+        return elicit_slot(session_attributes, intent_request, slots, 'Name',
+                           {'contentType': 'PlainText',
+                            'content': 'Sorry, did you say you name is {} ?'.format(" or ".join(resolved_names))})
+    else:
+        return continue_conversation(intent_request)
+
+
+def validate_month(intent_request):
+    """
+    Validate month is from an expected list, if not looks for other transcriptions and see if the month
+    recognized there has a value which is expected. If there is, replace with that and if not continue conversation
+    """
+
+    expected_months = ['january', 'february', 'march']
+    resolved_months = []
+    for transcription in intent_request['transcriptions']:
+        if transcription['resolvedSlots'] is not {} and 'BirthMonth' in transcription['resolvedSlots'] and \
+                transcription['resolvedSlots']['BirthMonth'] is not None:
+            resolved_months.append(transcription['resolvedSlots']['BirthMonth']['value']['originalValue'])
+
+    for resolved_month in resolved_months:
+        if resolved_month in expected_months:
+            intent_request['sessionState']['intent']['slots']['BirthMonth']['resolvedValues'] = [resolved_month]
+            break
+
+    return continue_conversation(intent_request)
+
+
+def continue_conversation(event):
```

Comment:
this seems redundant. Is the Lamnda expected to return empty response in case of fulfillment ? Shouldn't it return some valid message for customers to validate ?